lib/rules/no-use-computed-property-like-method.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @author tyankatsu <https://github.com/tyankatsu0105>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const { defineVueVisitor, getComputedProperties } = require('../utils')
+
+/**
+ * @typedef {import('../utils').ComponentComputedProperty} ComponentComputedProperty
+ * @typedef {import('../utils').ComponentObjectProp} ComponentObjectProp
+ */
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow use computed property like method',
+      categories: undefined,
+      url:
+        'https://eslint.vuejs.org/rules/no-use-computed-property-like-method.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected: 'Does not allow to use computed with this expression.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    /**
+     * @typedef {object} ScopeStack
+     * @property {ScopeStack | null} upper
+     * @property {BlockStatement | Expression} body
+     */
+    /** @type {Map<ObjectExpression, ComponentComputedProperty[]>} */
+    const computedPropertiesMap = new Map()
+
+    return defineVueVisitor(context, {
+      onVueObjectEnter(node) {
+        computedPropertiesMap.set(node, getComputedProperties(node))
```

Comment:
I think we need to make sure that the computed property is not a property that has a function.
To determine the type of a computed property, we need to track the variables in the return statement, as well as the types of `data` and `props`.
Also, properties whose types cannot be tracked should not be reported.

https://github.com/vuejs/eslint-plugin-vue/issues/928#issuecomment-653511036

You can use `findVariable` in `eslint-utils` to find the definition of a variable.
https://eslint-utils.mysticatea.dev/api/scope-utils.html#findvariable

You can use `getStaticValue` in `eslint-utils` to evaluate and get the value of an expression.
https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue

Reply:
👀 eslint-utils